truck_inspection_client/src/truck_inspection_client.cpp

```diff
@@ -79,6 +80,30 @@ namespace truck_inspection_client
     void TruckInspectionClient::requestCallback(const cav_msgs::MobilityRequestConstPtr& msg)
     {
         if(msg->strategy == this->INSPECTION_STRATEGY) {
+            std::uint16_t vin_retrive_count = 0;
+            //Checking: vin number does not exist yet in global parameter server
+            while((!pnh_->getParam("/vin_number", vin_number_)) && vin_retrive_count < MAX_RETRIEVE_VIN_COUNT)
+            {
+                //sleep for 0.1 second
+                boost::this_thread::sleep_for(boost::chrono::milliseconds(100));
+                vin_retrive_count++;
+
+                if(pnh_->getParam("/vin_number", vin_number_))
+                {
+                    ROS_INFO("retrieved vin_number1:  %s",vin_number_.c_str());
+                    break; 
+                }
+                ROS_WARN("retrieving vin_number:  %s",vin_number_.c_str());
```

Comment:
Lets make this a debug log. The info and error logs are really sufficient to describe what is happening.